cmd/metricsexporter/metricsExporter.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"flag"
+	"net"
+	"net/http"
+	"os"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/common/promlog"
+	promflag "github.com/prometheus/common/promlog/flag"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	exporterNamespace string
+	exporterAddress   string
+	kubeconfigPath    string
+	kubeAPIServerURL  string
+	verbose           bool
+)
+
+type lvmCollector struct{}
+
+func (lc *lvmCollector) Describe(c chan<- *prometheus.Desc) {}
+func (lc *lvmCollector) Collect(c chan<- prometheus.Metric) {}
+
+func main() {
+	lvmExpFlags := flag.NewFlagSet("lvm-metrics-exporter", flag.ExitOnError)
+	lvmExpFlags.StringVar(&exporterNamespace, "namespace", "lvm-exporter",
+		"set the namespace of the lvm metric exporter")
+	lvmExpFlags.StringVar(&exporterAddress, "address", ":23532",
+		"address on which the metrics exporter should run")
+	lvmExpFlags.StringVar(&kubeconfigPath, "kubeconfig", "", "Path to kubeconfig file")
+	lvmExpFlags.StringVar(&kubeAPIServerURL, "apiserver", "", "API server URL")
+	lvmExpFlags.BoolVar(&verbose, "verbose", false, "provide a verbose output")
+
+	lvmExpFlags.Parse(os.Args[1:])
+	newLogger := log.New()
+	newLogger.SetOutput(os.Stdout)
+	newLogger.SetFormatter(&log.TextFormatter{})
+	promlogConfig := &promlog.Config{}
+	promflag.AddFlags(kingpin.CommandLine, promlogConfig)
+	// promLogger := promlog.New(promlogConfig)
+	newLogger.WithField("args", os.Args[1:]).Info("LVM metrics exporter")
+	newLogger.Log(log.InfoLevel, "Namespace:", exporterNamespace)
+	newLogger.Log(log.DebugLevel, "Address:", exporterAddress)
+
+	// kubeConfig, err := clientcmd.BuildConfigFromFlags(kubeAPIServerURL, kubeconfigPath)
+	// if err != nil {
+	// 	logE.Fatal("Building kubeconfig from api-server URL: ", kubeAPIServerURL, " and kubeconfig: ", kubeconfigPath, ", failed. Error: ", err)
+	// }
+
+	prometheus.MustRegister(&lvmCollector{})
+
+	// prometheus http handler
+	http.Handle("/metrics", promhttp.Handler())
+	http.HandleFunc("/", func(rw http.ResponseWriter, r *http.Request) {
+		rw.Write([]byte(`<html>
+             <head><title>LVM Metric Exporter</title></head>
```

Comment:
(asking for my understanding) Why do we need this HTML ?

Reply:
This is added, if user try to access server without `/metrics` path appended in the address (like `127.0.0.1:8080/`).
This will guide them to reach correct location. It is a common practice seen in many metric exporters